File: C++/searching/binarySearch.cpp
//Binary Search

#include <iostream>
using namespace std;

int binarySearch(int arr[], int n, int k){
    int high = n-1;
    int low = 0;

    while(high>=low){
        int mid = (high + low)/2;
        if(arr[mid]==k) return mid;
        else if (k>arr[mid]){
            low = mid + 1;
        }
        else{
            high = mid - 1;
        }
    }
    return -1;
}

int main(){

    int size;
    cout<<"Enter the size of an array: ";
    cin>>size;
    int* arr = new int[size];

    for(int i=0;i<size;i++){
        cout<<"enter element "<<i<<": ";
        cin>>arr[i];
        cout<<endl;
    }

    int key;

    cout<<"Enter element you want to search: ";
    cin>>key;

    // int arr[] = {1,2,3,4,5};
    // int arr2[] = {12,23,45,65,100,101};
    cout<<"Element is present at index "<<binarySearch(arr, size, key)<<endl;
    // cout<<binarySearch(arr2, 6, 100)<<endl;
    // cout<<binarySearch(arr2, 6, 102)<<endl;
    // return 0;
}

File: Java/src/Searching/BinarySearch.java
package Searching;

import java.util.Scanner;

public class BinarySearch {

    static int binarySearch(int[] arr, int k){
        int n = arr.length;
        int high = n-1;
        int low = 0;
        while(low<=high){
            int mid = low + (high-low)/2;
            if(arr[mid]==k){
                return mid;
            }
            if(arr[mid]>k){
                high = mid - 1;
            }
            else{
                low = mid + 1;
            }
        }
        return -1;
    }

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        System.out.println("Enter size of array: ");
        int n = sc.nextInt();
        int[] arr = new int[n];
        System.out.println("Enter elements of Array: ");
        for(int i=0; i<n; i++){
            arr[i] = sc.nextInt();
        }
        System.out.println("Enter element to find: ");
        int key = sc.nextInt();

        System.out.println("Element found at: " + binarySearch(arr, key));
    }
}
